src/Symfony/Component/TypeInfo/Type/ArrayShapeType.php

```diff
@@ -81,7 +88,7 @@ public function accepts(mixed $value): bool
         foreach ($value as $key => $itemValue) {
             $valueType = $this->shape[$key]['type'] ?? false;
             if (!$valueType) {
-                return false;
+                return !$this->sealed;
```

Comment:
It should continue if it's not sealed, the type must be validated for each declared key.